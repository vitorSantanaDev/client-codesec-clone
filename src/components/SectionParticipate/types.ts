import { ISectionParticipateProps } from 'interfaces/sections'

export type IViewProps = ISectionParticipateProps
