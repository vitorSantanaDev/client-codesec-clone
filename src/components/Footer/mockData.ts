export const mockDataFooterProps = {
  logo: {
    altText: 'logo codesec',
    urlImage: '/images/logo-2-codesec.svg'
  },
  footerContent: 'Todos od direitos reservados',
  links: [
    {
      href: '/',
      icon: {
        altText: 'icon social',
        urlImage: '/images/instagram.svg'
      }
    },
    {
      href: '/',
      icon: {
        altText: 'icon social',
        urlImage: '/images/facebook.svg'
      }
    },
    {
      href: '/',
      icon: {
        altText: 'icon social',
        urlImage: '/images/youtube.svg'
      }
    }
  ]
}
