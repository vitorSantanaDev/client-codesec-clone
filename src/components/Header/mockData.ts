import { IViewProps } from './type'

export const mockHeaderProps: IViewProps = {
  logo: {
    urlImage: '/images/codesec-logo.svg',
    altText: 'Logo codesec'
  },
  button: {
    url: '/',
    buttonText: 'Inscreva-se'
  }
}
