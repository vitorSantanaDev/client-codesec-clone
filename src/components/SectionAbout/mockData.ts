export const mockSectionAboutProps = {
  subtitle: 'infos',
  title: 'About web series',
  description:
    'Aprenda PHP do Jeito Certo Utilizando 100% de Tecnologia Moderna e os Melhores Padr√µes de Projeto Nessa Webserie 100% Online e GRATUITA!',
  topics: [
    {
      _id: String(Math.random() * 1000),
      textInfo: 'text 1 example',
      icon: {
        altText: 'icone',
        urlImage: '/images/icon-eye.svg'
      }
    },
    {
      _id: String(Math.random() * 1000),
      textInfo: 'text 1 example',
      icon: {
        altText: 'icone',
        urlImage: '/images/icon-eye.svg'
      }
    },
    {
      _id: String(Math.random() * 1000),
      textInfo: 'text 1 example',
      icon: {
        altText: 'icone',
        urlImage: '/images/icon-eye.svg'
      }
    },
    {
      _id: String(Math.random() * 1000),
      textInfo: 'text 1 example',
      icon: {
        altText: 'icone',
        urlImage: '/images/icon-eye.svg'
      }
    }
  ],
  illustration: {
    urlImage: '/images/about-ilustration.svg',
    altText: 'Illustration about'
  }
}
