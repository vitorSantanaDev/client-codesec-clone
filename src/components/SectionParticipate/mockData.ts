import { IViewProps } from './types'

export const mockDataSection: IViewProps = {
  subtitle: 'Participe',
  title: 'Aprenda Webhacking',
  description:
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean imperdiet velit suscipit suscipit luctus. Mauris in enim erat. Cras sit amet cursus nunc.',
  illustration: {
    altText: 'Illustration',
    urlImage: '/images/ilustration-participe.svg'
  }
}
