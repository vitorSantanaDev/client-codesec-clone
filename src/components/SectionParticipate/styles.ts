import styled, { css } from 'styled-components'
import { Wrapper as Heading } from 'components/Heading/styles'
import { Wrapper as Subtitle } from 'components/Subtitle/styles'

export const Wrapper = styled.section`
  ${({ theme }) => css`
    width: 100%;
    display: flex;
    align-items: center;
    justify-content: space-between;

    padding: ${theme.spacings.huge} 0;

    @media ${theme.media.lteMedium} {
      flex-direction: column;
      justify-content: center;
    }
  `}
`

export const ContentWrapperLeft = styled.div`
  ${({ theme }) => css`
    > ${Subtitle} {
      margin-bottom: calc(${theme.spacings.small} - 0.8rem);
    }
    > ${Heading} {
      max-width: 433px;
      margin-bottom: calc(${theme.spacings.small} - 0.7rem);
    }
  `}
`

export const Description = styled.p`
  ${({ theme }) => css`
    max-width: 416px;
    color: ${theme.colors.gray7};
    font-size: ${theme.font.sizes.small};
    margin-bottom: ${theme.spacings.huge};
  `}
`

export const ContentWrapperRight = styled.div`
  ${() => css``}
`

export const Illustration = styled.img`
  ${({ theme }) => css`
    @media ${theme.media.lteMedium} {
      width: 350px;
      margin-top: ${theme.spacings.large};
    }
  `}
`
