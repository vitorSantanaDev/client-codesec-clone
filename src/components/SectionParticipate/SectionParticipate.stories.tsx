import { ComponentStory, ComponentMeta } from '@storybook/react'
import SectionParticipate from '.'
import { mockDataSection } from './mockData'

export default {
  title: 'SectionParticipate',
  component: SectionParticipate,
  args: {
    ...mockDataSection
  }
} as ComponentMeta<typeof SectionParticipate>

export const Default: ComponentStory<typeof SectionParticipate> = (args) => (
  <SectionParticipate {...args} />
)
