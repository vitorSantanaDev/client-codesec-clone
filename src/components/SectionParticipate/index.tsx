import Button from 'components/Button'
import Container from 'components/Container'
import Heading from 'components/Heading'
import Subtitle from 'components/Subtitle'

import { IViewProps } from './types'
import * as S from './styles'

const SectionParticipate = ({
  subtitle,
  title,
  description,
  illustration
}: IViewProps) => (
  <Container>
    <S.Wrapper>
      <S.ContentWrapperLeft data-aos="fade-up-right">
        <Subtitle>{subtitle}</Subtitle>
        <Heading>{title}</Heading>
        <S.Description>{description}</S.Description>
        <Button size="medium">Quero me cadastrar</Button>
      </S.ContentWrapperLeft>
      <S.ContentWrapperRight data-aos="fade-up-left">
        <S.Illustration
          src={illustration.urlImage}
          alt={illustration.altText}
        />
      </S.ContentWrapperRight>
    </S.Wrapper>
  </Container>
)

export default SectionParticipate
