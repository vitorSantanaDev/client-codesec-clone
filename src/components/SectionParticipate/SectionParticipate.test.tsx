import { screen } from '@testing-library/react'
import { renderTheme } from 'styles/theme/render-theme'
import SectionParticipate from '.'
import { mockDataSection } from './mockData'

describe('<SectionParticipate />', () => {
  it('should render the subtitle, Heading, and description', () => {
    renderTheme(<SectionParticipate {...mockDataSection} />)
    const subtitle = screen.getByText(mockDataSection.subtitle)
    const heading = screen.getByRole('heading', {
      name: `${mockDataSection.title}`
    })
    const description = screen.getByText(mockDataSection.description)

    expect(subtitle).toBeInTheDocument()
    expect(heading).toBeInTheDocument()
    expect(description).toBeInTheDocument()
  })

  it('should render the button with text correct', () => {
    renderTheme(<SectionParticipate {...mockDataSection} />)
    const button = screen.getByRole('button', { name: /quero me cadastrar/i })
    expect(button).toBeInTheDocument()
  })
  it('should render the illustration', () => {
    renderTheme(<SectionParticipate {...mockDataSection} />)
    const image = screen.queryByAltText(mockDataSection.illustration.altText)
    expect(image).toBeInTheDocument()
  })
  it('should match snapshot', () => {
    const { container } = renderTheme(
      <SectionParticipate {...mockDataSection} />
    )
    expect(container.firstChild).toMatchSnapshot()
  })
})
