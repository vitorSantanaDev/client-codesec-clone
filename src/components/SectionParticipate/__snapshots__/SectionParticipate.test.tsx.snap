// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SectionParticipate /> should match snapshot 1`] = `
.c8 {
  width: 100%;
  max-width: 255px;
  padding: calc(1.6rem - 0.4rem);
  font-weight: 600;
  text-transform: unset;
  font-size: 1.6rem;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  color: #FFFFFF;
  background: linear-gradient(243.46deg,#FA5770 16.66%,#E83752 83.34%);
  -webkit-transition: all ease-in-out 300ms;
  transition: all ease-in-out 300ms;
}

.c8:hover {
  -webkit-transform: scale(1.01);
  -ms-transform: scale(1.01);
  transform: scale(1.01);
  box-shadow: -10px 15px 40px -15px #FA5770;
}

.c0 {
  width: 100%;
  margin: 0 auto;
  max-width: 1200px;
  padding: 3.2rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c6 {
  font-size: 5.6rem;
  color: white;
  font-weight: 600;
  font-family: 'Raleway',sans-serif;
}

.c4 {
  background-image: linear-gradient(243.46deg,#FA5770 16.66%,#E83752 83.34%);
  background-clip: text;
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-weight: 700;
  -webkit-letter-spacing: 15%;
  -moz-letter-spacing: 15%;
  -ms-letter-spacing: 15%;
  letter-spacing: 15%;
  text-transform: uppercase;
  font-size: calc(1.6rem - 0.2rem);
}

.c1 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 5.6rem 0;
}

.c2 > .c3 {
  margin-bottom: calc(1.6rem - 0.8rem);
}

.c2 > .c5 {
  max-width: 433px;
  margin-bottom: calc(1.6rem - 0.7rem);
}

.c7 {
  max-width: 416px;
  color: #A3A3A3;
  font-size: 1.6rem;
  margin-bottom: 5.6rem;
}

@media (max-width:768px) {
  .c6 {
    font-size: 4.0rem;
  }
}

@media (max-width:768px) {
  .c1 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }
}

@media (max-width:768px) {
  .c9 {
    width: 350px;
    margin-top: 3.2rem;
  }
}

<div
  class="c0"
>
  <section
    class="c1"
  >
    <div
      class="c2"
      data-aos="fade-right"
    >
      <span
        class="c3 c4"
        color="redGradient"
      >
        Participe
      </span>
      <h1
        class="c5 c6"
        color="white"
      >
        Aprenda Webhacking
      </h1>
      <p
        class="c7"
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean imperdiet velit suscipit suscipit luctus. Mauris in enim erat. Cras sit amet cursus nunc.
      </p>
      <button
        class="c8"
      >
        Quero me cadastrar
      </button>
    </div>
    <div
      class=""
      data-aos="fade-up"
    >
      <img
        alt="Illustration"
        class="c9"
        src="/images/ilustration-participe.svg"
      />
    </div>
  </section>
</div>
`
